dolphin/task_manager/tasks/task.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+
+from dolphin import exception
+from dolphin.drivers import manager as drivermanager
+from dolphin.db.sqlalchemy import api as db
+from dolphin.i18n import _
+
+LOG = log.getLogger(__name__)
+
+
+class StorageResourceTask(object):
+
+    def __init__(self, context, storage_id):
+        self.storage_id = storage_id
+        self.context = context
+        self.driver = drivermanager.DriverManager()
+
+
+class StorageDeviceTask(StorageResourceTask):
+    def __init__(self, context, storage_id):
+        super(StorageDeviceTask, self).__init__(context, storage_id)
+
+    def sync(self):
+        """
+        :return:
+        """
+        LOG.info('syncing storage device for storage id:{0}'.format(self.storage_id))
```

Comment:
Would this be called in periodic task?
If yes, the log at the beginning and at the end would flush the log.